src/chemistry/network/kida.cpp

```diff
@@ -0,0 +1,2315 @@
+      rids.push_back(ri.id_);
+      if (ri.id_ > id7max_ && ri.formula_ == 7) {
+        id7max_ = ri.id_;
+      }
+      reactions_.push_back(ri);
+    } else {
+      if (reactions_[ifind].reactants_ == ri.reactants_
+          && reactions_[ifind].products_ == ri.products_
+          && reactions_[ifind].itype_ == ri.itype_
+          && reactions_[ifind].Tmax_ < ri.Tmin_
+          && reactions_[ifind].formula_ != 7 && ri.formula_ != 7
+          && ri.itype_ >= 4 && ri.itype_<= 8 ) {
+        if (ifind != nline - 1) {
+          std::stringstream msg; // error message
+          msg << "### FATAL ERROR in ChemNetwork constructor [ChemNetwork]"
+            << std::endl << "reactions ID ( " << ri.id_ << " ) of different"
+            << " temperature ranges are not arranged next to each other."
+            << std::endl;
+          ATHENA_ERROR(msg);
+        }
+        rids.push_back(ri.id_);
+        reactions_.push_back(ri);
+        const int nprev = std::count(id_2bodytr_.begin(), id_2bodytr_.end(), ri.id_);
+        if (nprev == 0) {
+          id_2bodytr_.push_back(ri.id_);
+          n_2bodytr_++;
+        } else if (nprev < n_range_ - 1) {
+          id_2bodytr_.push_back(ri.id_);
+        } else {
+          std::stringstream msg; // error message
+          msg << "### FATAL ERROR in ChemNetwork constructor [ChemNetwork]"
+            << std::endl << "reaction ID ( " << ri.id_ << " )"
+            << "too many temperature ranges" << std::endl;
+          ATHENA_ERROR(msg);
+        }
+      } else {
+        std::stringstream msg; // error message
+        msg << "### FATAL ERROR in ChemNetwork constructor [ChemNetwork]"
+          << std::endl << "reaction ID ( " << ri.id_ << ") not unique" << std::endl;
+        ATHENA_ERROR(msg);
+      }
+    }
+    nline++;
+  }
+  nr_ = nline; // number of reactions
+  if (id7max_ > 0) {
+    id7map_.NewAthenaArray(id7max_+1);
+    id7type_.NewAthenaArray(id7max_+1);
+    for (int i=0; i<id7max_+1; i++) {
+      id7map_(i) = -1;
+      id7type_(i) = ReactionType::none;
+    }
+  }
+
+  // initialize coefficients of reactions
+  InitializeReactions();
+
+  // radiation related variables
+  const int nfreq = pin->GetOrAddInteger("chem_radiation", "n_frequency", 1);
+  n_freq_ = n_ph_ + 2;
+  std::stringstream msg;
+  // check whether number of frequencies equal to the input file specification
+  if (nfreq != n_freq_) {
+    msg << "### FATAL ERROR in ChemNetwork constructor" << std::endl
+      << "number of frequencies in radiation: " << nfreq
+      << " not equal to that in chemistry: " << n_freq_  << std::endl;
+    ATHENA_ERROR(msg);
+  }
+  index_gpe_ = n_ph_;
+  index_cr_ = n_ph_ + 1;
+  rad_.NewAthenaArray(n_freq_);
+
+#ifdef DEBUG
+  PrintProperties();
+#endif
+}
+
+//----------------------------------------------------------------------------------------
+//! \brief ChemNetwork destructor
+ChemNetwork::~ChemNetwork() {
+  FILE *pf = fopen("chem_network.dat", "w");
```

Comment:
All of this should be removed from the destructor. I get tons of Valgrind errors when the class object goes out of scope

Reply:
Removed.